File: TestSamples/Haskell_Samples/Where6.hs
module Main where

g z = z+(let h=1 in h+5)+((\x -> x*x+s)3)+(let h=1 in h+5)
where s = (let h=1 in h+5)+((\x -> x*x+((\z->z*z)5))3)

main = do { 
              (print(g 1));
          }

File: TestSuite/where6.pas
type
  t0=record end;
  
  t1<T1>=class
  where T1: record;
  a: T1;
  end;
  
  t2<T>=class(Dictionary<t1<t0>,T>) where T: record; end;

begin
  var a := new t2<integer>;
  var o := new t1<t0>;
  a.Add(o, 2);
  assert(a[o] = 2);
end.